api/src/test/java/org/openmrs/DrugOrderTest.java

```diff
@@ -314,4 +314,100 @@ public void setAutoExpireDateBasedOnDuration_shouldNotCalculateIfAutoExpireDateA
 		
 		assertEquals(expectedAutoExpireDate, drugOrder.getAutoExpireDate());
 	}
+
+	/**
+	 * @verifies return true if the drugs match and drug non coded match
+	 * @see DrugOrder#hasSameOrderableAs(Order)
+	 */
+	@Test
+	public void hasSameOrderableAs_shouldReturnTrueIfTheConceptsMatchAndHaveSameDrugNonCoded() throws Exception {
+		DrugOrder order = new DrugOrder();
+		Concept concept = new Concept();
+		Drug drug1 = new Drug();
+		drug1.setConcept(concept);
+		order.setDrug(drug1);
+		order.setDrugNonCoded("Chrocine");
```

Comment:
This clarifies what i was saying above, do you realize that you always have to set drug for a non coded order? Meaning isNoneCodedDrug technically can never return true if itself drug is null